### PR TITLE
feat(type): wrap @font-face in mix-in

### LINKED ISSUE #143: Type: Request for wrapping @font-face in mix-in
**Is your feature request related to a problem? Please describe.**

Initial look at IBM/carbon-components#1546 revealed that `@font-face` gets effective with `@import '@carbon/type/scss/font-face/sans'`, etc. regardless of how consuming application wants to set things up. (e.g. Choosing CDN to serve fonts)

Another finding is that it seems hard to make `@font-face` effective only when `carbon-components`' `breaking-changes-x` feature flag is `true`.

**Describe the solution you'd like**

This issue requests for wrapping `@font-face`s in mix-ins, similar to what `carbon-components` does: https://github.com/IBM/carbon-components/blob/v9.0.0/src/globals/scss/_css--plex-core.scss#L37-L52

**Describe alternatives you've considered**

An alternative is maintaining its own definition of `@font-face` in `carbon-components` library, similar to what `carbon-components` did for initial Plex implementation.

### PR DESCRIPTION
**Pull request by asudoh**
Fixes #143.

#### Changelog

**Changed**

- Wrap `@font-face`s in explicit mix-ins, instead of import-once guard.

### CODE REVIEW

File: packages/type/scss/font-face/_mono.scss [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 @import '@carbon/import-once/scss/import-once';
```
Comment (joshblack) [inline packages/type/scss/font-face/_mono.scss]: Could we remove the `import-once` import since we aren't using it?

Comment (asudoh) [reply packages/type/scss/font-face/_mono.scss]:
```suggestion
```

File: packages/type/scss/font-face/_mono.scss [hunk 2]
```diff
@@ -5,9 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
```
Comment (asudoh) [inline packages/type/scss/font-face/_mono.scss]:
```suggestion
```

**Comment by asudoh** [comment]
Oops good catch @joshblack! - Fixed.

**Comment by joshblack** [comment]
@asudoh what do you think of adding the file import alongside the mixin? it's been nice to be able to just:

```
@import '@carbon/type/scss/font-family/sans';
```

And have sans loaded in automatically

**Comment by asudoh** [comment]
@joshblack If I understand what you said correctly - Carbon has `$css--font-face` variable for that purpose. Good to have a similar mechanism here, though we cannot use the same name because Carbon has to choose what library to load `@font-face`.

**Comment by joshblack** [comment]
@asudoh I think we should offer:

- Mixin for conditional inclusion (used to satisfy flag cases like for carbon)
- File import for ease-of-use for people using directly

**Comment by joshblack** [comment]
CI is failing on the `bundler check` command, just need to add in an import for the `exports` mixin so the individual partial compiles! 👍

**Comment by asudoh** [comment]
Thanks @joshblack for the update!

**Pull request merged**